e2e_playwright/st_arrow_altair_chart_test.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+def test_displays_altair_chart(themed_app: Page):
+    expect(
+        themed_app.get_by_test_id("stArrowVegaLiteChart").locator("canvas")
+    ).to_have_count(8)
+
+
+def test_altair_chart_displays_correctly(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    charts = themed_app.get_by_test_id("stArrowVegaLiteChart")
+    expect(charts).to_have_count(8)
+    for index in range(charts.count()):
```

Comment:
nit: it think we can move the check above just to down here and remove the other test method. Thereby, even lower risk for flakiness. Reason is: the canvas check is a waiting check (waiting for the canvas elements to appear) while the assert_snapshot is not. So, potentially there could be a situation where we do a snapshot before the canvas is even loaded. If this actually makes a difference depends a bit on the internal of vega lite chart.

```suggestion
    expect(
        themed_app.get_by_test_id("stArrowVegaLiteChart").locator("canvas")
    ).to_have_count(8)
    
    for index in range(charts.count()):
```